### PR TITLE
mds: don't maintain bloom filters in standby replay

### PR DESCRIPTION
**Pull request by jcsp**
Fixes: http://tracker.ceph.com/issues/16924

### CODE REVIEW

File: src/vstart.sh [hunk 1]
```diff
@@ -761,6 +757,20 @@ EOF
     done
 fi
 
+# Don't set max_mds until all the daemons are started, otherwise
+# the intended standbys might end up in active roles.
+fs=0
+for name in a b c d e f g h i j k l m n o p
+do
+    sleep 5  # wait for daemons to make it into FSMap
```
Comment (gregsfortytwo) [inline src/vstart.sh]: This appears to be sleeping 5 seconds for each fs. I don't think that was meant to happen?

File: src/mds/journal.cc [hunk 2]
```diff
@@ -2849,6 +2849,27 @@ void dirfrag_rollback::decode(bufferlist::iterator &bl)
...
+  dir->remove_bloom();
+
+  std::list<CDir*> subdirs;
+  CDir::map_t::iterator it;
+  for (it = dir->begin(); it != dir->end(); ++it) {
+    CDentry *dn = it->second;
+    CInode *in = dn->get_linkage()->get_inode();
+
+    if (dn->get_linkage()->is_primary()) {
+      in->get_nested_dirfrags(subdirs);
```
Comment (gregsfortytwo) [inline src/mds/journal.cc]: I must be misremembering how EExport works. Why do we need to go down one level to pull out the child CDir bloom filters, but not all the way to the leaves?

**Comment by ukernel** [comment]
I think it's better to avoid adding bloom filter while standby replay

File: src/mds/journal.cc (hunk 2 above)
Comment (ukernel) [inline src/mds/journal.cc]: remove_blooms calls itself, it does go down to the leaves.

**Comment by jcsp** [comment]
@ukernel good point, I agree.  I've updated this.

**Comment by gregsfortytwo** [comment]
Reviewed-by: Greg Farnum <gfarnum@redhat.com>

**Pull request merged**